apps/scroll-cd/src/app/app.component.ts

```diff
@@ -31,16 +40,42 @@ import { BehaviorSubject } from 'rxjs';
     `,
   ],
 })
-export class AppComponent {
+export class AppComponent implements OnDestroy, OnInit {
   title = 'scroll-cd';
 
-  private displayButtonSubject = new BehaviorSubject<boolean>(false);
-  displayButton$ = this.displayButtonSubject.asObservable();
+  destroy = new Subject();
+  destroy$ = this.destroy.asObservable();
+  displayButton = false;
 
-  @HostListener('window:scroll', ['$event'])
-  onScroll() {
-    const pos = window.pageYOffset;
-    this.displayButtonSubject.next(pos > 50);
+  constructor(
+    @Inject(DOCUMENT)
+    private document: { documentElement: { pageYOffset: number } },
+    private readonly zone: NgZone
+  ) {}
+
+  ngOnInit(): void {
+    this.zone.runOutsideAngular(() => {
+      fromEvent(window, 'scroll')
+        .pipe(
+          map(
+            () =>
+              window.pageYOffset ||
+              this.document.documentElement.pageYOffset ||
+              0
+          ),
+          map((pageYOffset: number) => pageYOffset > 50),
+          distinctUntilChanged(),
+          takeUntil(this.destroy$),
+          map((value: boolean) =>
+            this.zone.run(() => (this.displayButton = value))
```

Comment:
The function in a `map` should be a pure function. This is using a map solely for a side effect. This behavior would belong in a `tap`, or even better the `subscribe` that you already have.